libs/process/forwarded.go

```diff
@@ -0,0 +1,42 @@
+package process
+
+import (
+	"context"
+	"io"
+	"os/exec"
+	"strings"
+
+	"github.com/databricks/cli/libs/log"
+)
+
+func Forwarded(ctx context.Context, args []string, src io.Reader, dst io.Writer, opts ...execOption) error {
+	commandStr := strings.Join(args, " ")
+	log.Debugf(ctx, "starting: %s", commandStr)
+	cmd := exec.CommandContext(ctx, args[0], args[1:]...)
+
+	// make sure to sync on writing to stdout
+	reader, writer := io.Pipe()
+
+	// empirical tests showed buffered copies being more responsive
+	go io.CopyBuffer(dst, reader, make([]byte, 128))
+	defer reader.Close()
+	defer writer.Close()
+	cmd.Stdout = writer
+	cmd.Stderr = writer
```

Comment:
As discussed; the copying is not needed if you add a `out, err io.Writer` and set them on `cmd` directly.

Then the caller can pass in `os.Stdout` and `os.Stderr` and Go is no longer in the loop, while you can still do unit testing by passing in `bytes.Buffer` instances.